restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -803,28 +798,70 @@ void addRowsToMap(Map<String, List<Row>> rowsByDoc, List<Row> rows) {
     return Collections.emptyList();
   }
 
+  /**
+   * If the DocumentSearchPageState has a defined last seen `documentId`, advance the result set to
+   * be just after that ID.
+   */
+  private List<Row> skipSeenRows(DocumentSearchPageState initialPagingState, List<Row> rows) {
+    if (initialPagingState == null) {
+      return rows;
+    }
+
+    String lastSeenId = initialPagingState.getLastSeenDocId();
+    if (lastSeenId == null) {
+      return rows;
+    }
+
+    boolean idFound = false;
+    boolean doneSkipping = false;
+    while (!doneSkipping && rows.size() > 0) {
+      String docId = rows.get(0).getString("key");
+      if (docId.equals(lastSeenId)) {
+        idFound = true;
+        rows.remove(0);
```

Comment:
Removing from iterators may not be efficient and may not even be supported if the source is not a plain list.

Would we use filtering instead?